automation/wdio.conf.js

```diff
@@ -0,0 +1,241 @@
+    //
+    // Default timeout for all waitFor* commands.
+    waitforTimeout: 30000,
+    //
+    // Default timeout in milliseconds for request
+    // if browser driver or grid doesn't send response
+    connectionRetryTimeout: 120000,
+    //
+    // Default request retries count
+    connectionRetryCount: 3,
+    framework: 'cucumber',
+    reporters: [['allure', {
+        outputDir: 'allure-results',
+        disableWebdriverStepsReporting: true,
+        disableWebdriverScreenshotsReporting: false,
+        useCucumberStepReporter: true
+    }]],
+    cucumberOpts: {
+        // <string[]> (file/dir) require files before executing features
+        require: ['./features/step-definitions/*.steps.js'],
+        // <boolean> show full backtrace for errors
+        backtrace: false,
+        // <string[]> ("extension:module") require files with the given EXTENSION after requiring MODULE (repeatable)
+        requireModule: [],
+        // <boolean> invoke formatters without executing steps
+        dryRun: false,
+        // <boolean> abort the run on first failure
+        failFast: false,
+        // <boolean> hide step definition snippets for pending steps
+        snippets: true,
+        // <boolean> hide source uris
+        source: true,
+        // <boolean> fail if there are any undefined or pending steps
+        strict: false,
+        // <string> (expression) only execute the features or scenarios with tags matching the expression
+        tagExpression: 'not @todo and not @manual and not @ignore',
+        // <number> timeout for step definitions
+        timeout: 30000,
+        // <boolean> Enable this config to treat undefined definitions as warnings.
+        ignoreUndefinedDefinitions: false
+    },
+
+    before: function (capabilities, specs) {
+        // Add commands to WebdriverIO
+          Object.keys(commands).forEach(key => {
+              console.log('Adding custom command - ',key)
+          browser.addCommand(key, commands[key]);
+              })
+          },
+    // beforeFeature: function (uri, feature){
+    //     Hooks.connectEthWallet()
+    //     },
+
+    afterStep: function (step, scenario, result, context) {
+      if (!result.passed){
+        browser.takeScreenshot()
+      }
+    },
+
+    onComplete: function() {
+        const reportError = new Error('Could not generate Allure report')
+        const generation = allure(['generate', 'allure-results', '--clean'])
+        return new Promise((resolve, reject) => {
+            const generationTimeout = setTimeout(
+                () => reject(reportError),
+                30000)
+
+            generation.on('exit', function(exitCode) {
+                clearTimeout(generationTimeout)
+
+                if (exitCode !== 0) {
+                    return reject(reportError)
+                }
+                resolve()
+            })
+        })
+    }
+    /**
+     * Runs before a WebdriverIO command gets executed.
```

Comment:
Can we clean up the unused config below? (same for the other configs)

Reply:
ideally nice to keep for readability if we want to add extra configs but i've commented this out